eos-phone-home

```diff
@@ -277,6 +281,54 @@ class PhoneHome(object):
 
         return False
 
+    def _get_mac_hash(self):
+        mac_hash = None
+
+        try:
+            ifaces = []
+            wireless = []
+
+            for iface in glob.glob('/sys/class/net/*'):
+                # skip virtual interfaces
+                device = os.path.join(iface, 'device')
+                if not os.path.exists(device):
+                    continue
+
+                # skip USB devices
+                subsystem = os.readlink(os.path.join(device, 'subsystem'))
+                if os.path.basename(subsystem) == 'usb':
+                    continue
+
+                # build separate list of wireless interfaces
+                if os.path.exists(os.path.join(iface, 'wireless')):
+                    wireless.append(iface)
+                else:
+                    ifaces.append(iface)
+
+            # prefer the fixed interfaces if there are any, wireless if not
+            if len(ifaces) == 0:
+                ifaces = wireless
+
+            if len(ifaces) == 0:
+                return mac_hash
```

Comment:
I think it'd be clearer to `return None` here. I appreciate you're doing this for consistency with how `None` is returned in case of an exception but I'd personally prefer an explicit `return None` in the `except` block and no forward declaration of `mac_hash`. Opinions may differ…

Reply:
I actually had it the other way when I was implementing it, but the other get functions do early define and (typically) unconditional return. This one has an early return so... I've taken away the forward declaration.